pkg/sharding/sharding.go

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sharding
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/sigstore/rekor/pkg/api"
+)
+
+// A FullID refers to a specific artifact's ID and is made of two components,
+// the TreeID and the UUID. The TreeID is a hex-encoded uint64 (8 bytes)
+// referring to the specific trillian tree (also known as log or shard) where
+// the artifact can be found. The UUID is a hex-encoded 32-byte number
+// referring to the artifact's merkle leaf hash from trillian. Artifact lookup
+// by UUID occurs by finding the UUID within the tree specified by the TreeID.
+//
+// A FullID is 40 bytes long and looks like this:
+// FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF
+// |_______  ________| |_____________________________________  ______________________________________|
+//         \/                                                \/
+// TreeID (8 bytes, hex)                             UUID (32 bytes, hex)
+
+const TreeIDHexStringLen = 16
+const UUIDHexStringLen = 64
+const FullIDHexStringLen = TreeIDHexStringLen + UUIDHexStringLen
+
+// TODO: replace this with the actual LogRanges struct when logic is hooked up
+var dummy = api.LogRanges{
+	Ranges: []api.LogRange{},
+}
+
+type FullID struct {
+	TreeID string
+	UUID   string
+}
+
+func CreateFullID(treeid string, uuid string) (FullID, error) {
+	if len(treeid) != TreeIDHexStringLen {
+		err := fmt.Errorf("invalid treeid len: %v", len(treeid))
+		return CreateEmptyFullID(), err
+	}
+
+	if len(uuid) != UUIDHexStringLen {
+		err := fmt.Errorf("invalid uuid len: %v", len(uuid))
+		return CreateEmptyFullID(), err
+	}
+
+	return FullID{
+		TreeID: treeid,
+		UUID:   uuid}, nil
+}
+
+func CreateEmptyFullID() FullID {
```

Comment:
is this intent for this to be public or just a internal helper function?

Reply:
Good catch, I'm still getting used to go's use of caps to mean public. I suppose for now it should be internal so I'll update it.